File: lambdas/updateTask.js
const AWS = require('aws-sdk');
const dynamo = new AWS.DynamoDB.DocumentClient();
 
exports.handler = async (event) => {
  const { id } = event.pathParameters;
  const data = JSON.parse(event.body);
 
  const params = {
    TableName: 'ToDoTable',
    Key: { id },
    UpdateExpression: 'set task = :task, status = :status',
    ExpressionAttributeValues: {
      ':task': data.task,
      ':status': data.status
    },
    ReturnValues: 'UPDATED_NEW'
  };
 
  await dynamo.update(params).promise();
  return {
    statusCode: 200,
    body: JSON.stringify({ message: 'Task updated' })
  };
};

File: lambda/updateTask.py
import json
import boto3

dynamodb = boto3.resource('dynamodb')
table = dynamodb.Table('Tasks')

def lambda_handler(event, context):
    task_id = event['pathParameters']['taskId']
    data = json.loads(event['body'])

    table.update_item(
        Key={'taskId': task_id},
        UpdateExpression="set title = :t, #s = :s",
        ExpressionAttributeValues={
            ':t': data['title'],
            ':s': data['status']
        },
        ExpressionAttributeNames={
            '#s': 'status'
        }
    )

    return {
        'statusCode': 200,
        'body': json.dumps({'message': 'Task updated'})
    }
